src/api/roles/coach/coach.py

```
@@ -65,9 +66,83 @@ def create_coach_request(coach_details: CoachRequestInput, db = Depends(get_sess
+    Errors when user does not have a coach_id
+    """
+    if acc.coach_id is None:
+        raise HTTPException(404, detail="No coach profile found for this account")
+    
+    db.query(Coach).filter(Coach.id == acc.coach_id).delete(synchronize_session=False)
+
+    acc.coach_id = None
+    
+    db.commit()
+
+    return CoachRequestDeniedResponse(coach_request_id=cr.id, coach_id=acc.coach_id) # type: ignore
```

Ruff (0.15.9)

[error] Undefined name `cr`

(F821)